### PR TITLE
feat(Google Photos): add presence

### PR DESCRIPTION
**Pull request by kyrie25**
#### Resolves discussion #4482

<details>
<summary> Proof of the presence working: </summary>
<br>

image
image
image
image
image

</details>

### CODE REVIEW

**Comment by kyrie25** [comment]
#### Add settings for `Show Album Name`
<details>
<summary> Modifications: </summary>

image
image
image

</details>

**Review by Bas950 — DISMISSED** [review]

File: websites/G/Google Photos/presence.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    presenceData.details = "Viewing a photo";
+  else if (path.startsWith("/search")) {
+    presenceData.details = "Searching for:";
+    [presenceData.state] = document.title.split(/-/, 1);
+  } else if (path === "/albums") presenceData.details = "Browsing albums";
+  else if (path.startsWith("/album")) {
+    presenceData.details = "Viewing an album:";
+    presenceData.state = (await presence.getSetting<boolean>("albumname"))
+      ? document.title.split(/-/, 1)[0]
+      : "(Hidden in presence settings)";
```
Comment (EncryptedDev) [inline websites/G/Google Photos/presence.ts:26]: Just use an if and don't set state if the setting is false

Comment (kyrie25) [reply websites/G/Google Photos/presence.ts]: done

File: websites/G/Google Photos/presence.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  else if (path.startsWith("/search")) {
+    presenceData.details = "Searching for:";
+    [presenceData.state] = document.title.split(/-/, 1);
+  } else if (path === "/albums") presenceData.details = "Browsing albums";
+  else if (path.startsWith("/album")) {
+    presenceData.details = "Viewing an album";
+    {
+      if (await presence.getSetting<boolean>("albumname"))
+        [presenceData.state] = document.title.split(/-/, 1);
+      else presenceData.state;
```
Comment (EncryptedDev) [inline websites/G/Google Photos/presence.ts:27]:
```suggestion
```

File: websites/G/Google Photos/presence.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    }
+  } else if (path.startsWith("/archive"))
+    presenceData.details = "Viewing the archive";
+  else presenceData;
```
Comment (EncryptedDev) [inline websites/G/Google Photos/presence.ts:31]:
```suggestion
```

**Review by EncryptedDev — DISMISSED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Review by Slowlife01 — APPROVED** [review]

**Pull request merged**